File: C#/TreeSearch.cs
using System;

public class TreeSearch
{
    private class TreeNode
    {
        public int Value;
        public TreeNode Left;
        public TreeNode Right;

        public TreeNode(int value)
        {
            Value = value;
            Left = Right = null;
        }
    }

    private TreeNode root;

    public TreeSearch()
    {
        root = null;
    }

    //Einfügen in den Baum
    public void Insert(int value)
    {
        if (root == null)
        {
            root = new TreeNode(value);
        }
        else
        {
            InsertRec(root, value);
        }
    }

    private void InsertRec(TreeNode node, int value)
    {
        if (value < node.Value)
        {
            if (node.Left == null)
                node.Left = new TreeNode(value);
            else
                InsertRec(node.Left, value);
        }
        else
        {
            if (node.Right == null)
                node.Right = new TreeNode(value);
            else
                InsertRec(node.Right, value);
        }
    }

    //Suchen im Baum
    public bool Search(int value)
    {
        return SearchRec(root, value);
    }

    private bool SearchRec(TreeNode node, int value)
    {
        if (node == null) return false;
        if (node.Value == value) return true;
        return value < node.Value ? SearchRec(node.Left, value) : SearchRec(node.Right, value);
    }

    public static void Main(string[] args)
    {
        TreeSearch tree = new TreeSearch();
        tree.Insert(10);
        tree.Insert(5);
        tree.Insert(15);
        tree.Insert(3);
        tree.Insert(7);

        Console.WriteLine("Suche 5: " + tree.Search(5));   
        Console.WriteLine("Suche 7: " + tree.Search(7));   
        Console.WriteLine("Suche 20: " + tree.Search(20));  
        Console.WriteLine("Suche 10: " + tree.Search(10));  
        Console.WriteLine("Suche 3: " + tree.Search(3));   
    }
}

File: Python/Collection/TreeSearch.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

# Binärer Suchbaum
class TreeSearch:
    def __init__(self):
        self.root = None

    # Wert hinzufügen
    def insert(self, value):
        if self.root is None:
            self.root = TreeNode(value)
        else:
            self._insert_rec(self.root, value)

    def _insert_rec(self, node, value):
        if value < node.value:
            if node.left is None:
                node.left = TreeNode(value)
            else:
                self._insert_rec(node.left, value)
        else:
            if node.right is None:
                node.right = TreeNode(value)
            else:
                self._insert_rec(node.right, value)

    # Wert suchen
    def search(self, value):
        return self._search_rec(self.root, value)

    def _search_rec(self, node, value):
        if node is None:
            return False
        if node.value == value:
            return True
        return self._search_rec(node.left, value) if value < node.value else self._search_rec(node.right, value)

# Beispielnutzung
if __name__ == "__main__":
    tree = TreeSearch()
    tree.insert(10)
    tree.insert(5)
    tree.insert(15)
    tree.insert(3)
    tree.insert(7)

    print("Suche 5:", tree.search(5))  # Ausgabe: True
    print("Suche 7:", tree.search(7))  # Ausgabe: True
    print("Suche 20:", tree.search(20)) # Ausgabe: False
    print("Suche 10:", tree.search(10)) # Ausgabe: True
    print("Suche 3:", tree.search(3))  # Ausgabe: True
